runtime/major_gc.c

```diff
@@ -368,10 +368,18 @@ static intnat ephe_mark (intnat budget, uintnat for_cycle,
     }
     budget -= Whsize_wosize(i);
 
-    if (force_alive || alive_data) {
-      if (data != caml_ephe_none && Is_block(data)) {
-        caml_darken (domain_state, data, 0);
-      }
+    if (data == caml_ephe_none || Is_long(data)) {
+      /* Not yet known whether this ephemeron's keys/block will be marked,
+         but since the data is trivial nothing will happen if they are,
+         so remove it from the todo list */
+      Ephe_link(v) = domain_state->ephe_info->live;
```

Comment:
It would be nice if: this duplicated code weren't duplicated.

Reply:
Fixed